src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs

```diff
@@ -39,6 +39,12 @@ protected override UInt256 GetKeyForItem((UInt256, DateTime) item)
 
         private void OnMessage(Message msg)
         {
+            if (msg.Command == MessageCommand.Extensible)
+            {
+                var payload = (ExtensiblePayload)msg.Payload;
+                if (!knownHashes.Add(payload.Hash)) return;
+                if (!payload.Verify(Blockchain.Singleton.GetSnapshot())) return;
+            }
```

Comment:
It is not necessary. The payload will be verified later.

Reply:
We should verify it before the plugin processes it, isn't it?

Reply:
Maybe we should modify `IP2PPlugin` to allow plugins to receive verified messages.

Reply:
Please check it